### ISSUE TITLE
Logging not working

### ISSUE THREAD

**Issue by archerian** [emqx/emqx#2546]
#### Environment

- OS: Windows 10
- Erlang/OTP: 21
- EMQ: 3.1.1

#### Description

Logging to file is not working, emqx setting is set to
```properties
##--------------------------------------------------------------------
## Log
##--------------------------------------------------------------------

## Where to emit the logs.
## Enable the console (standard output) logs.
##
## Value: off | file | console | both
## - off: disable logs entirely
## - file: write logs only to file
## - console: write logs only to standard I/O
## - both: write logs both to file and standard I/O
log.to = file

## The log severity level.
##
## Value: debug | info | notice | warning | error | critical | alert | emergency
##
## Note: Only the messages with severity level higher than or equal to
##       this level will be logged.
##
## Default: error
log.level = debug

## The dir for log files.
##
## Value: Folder
log.dir = log

## The log filename for logs of level specified in "log.level".
##
## Value: String
## Default: emqx.log
log.file = emqx.log

```

**Comment by MikeRavenelle** [comment]
Is there a last known version where logging was functional?

**Comment by terry-xiaoyu** [comment]
3.1.0

**Comment by terry-xiaoyu** [comment]
@MikeRavenelle For the log level issue in 3.1.1. A workaround is using the `emqx_ctl log` CLI. See
https://developer.emqx.io/docs/broker/v3/en/commands.html#log

**Comment by HJianBo** [comment]
Hi, @MikeRavenelle We have released this fix at v3.1.2. Please check it

Let me know please, if it can not work

Thanks
